process-automation/latest/model/connectors/docusign.md

```diff
@@ -2,87 +2,118 @@
-"ahr:role": "Developer"
-}
-```
+The **DOWNLOAD_DOCUMENT** action is used by the DocuSign connector to request a digital signature on a document.
 
-The output parameters from signing a document are:
+The input parameters for DOWNLOAD_DOCUMENT are:
 
 | Parameter | Type | Description |
 | --------- | ---- | ----------- |
-| file | File | *Optional.* The signed document available to be mapped to a [variable]({% link process-automation/latest/model/processes/index.md %}#process-variables). |
+| outputFileName | String | *Optional.* The name of the file that will be created, for example `invoice.pdf`. |
+| envelopeId | yes | *Required* UUID related to the envelope from the DocuSign API. |
+| targetFolder | Folder | *Requires one.* A [variable]({% link process-automation/latest/model/processes/index.md %}#process-variables) of type folder to store the signed document in. |
+| targetFolderId | String | *Optional.* The target folder ID to save the document in Content Services. |
+| targetFolderPath | String | *Requires one.* The location path or relative path of the folder to store the signed document in Content Services. For example, a location path: `/app:company_home/app:user_homes/cm:hruser` or a relative path: `/User Homes/hruser`.  |
+| targetFileType | String | *Optional.* The type to set for the signed file, for example `fin:invoice`. |
+| targetFileMetadata | content-metadata | *Optional.* Metadata assigned to the signed document in Content Services. |
+| underscoreMetadata | boolean | *Optional.* If set to `true` the received prefixed properties names contain underscores (_) instead of colons (:), for separating the namespace prefix from the property name. |
+
+The output parameters from DOWNLOAD_DOCUMENT are:
+
+| Property | Type | Description |
+|--- | --- | --- |
+| file | file | The file stored in Content Services of the document. |
+
+No exceptions or errors are thrown by the connector, however all exceptions are caught and logged. The task execution is always successful, and errors will be returned in an error event.
+
+### Events
+
+The DocuSign connector produces events when the DocuSign envelope changes its status, the events are:
+
+* `ENVELOPE_VOIDED` -  The envelope has been voided by the sender or has expired. The void reason indicates whether the envelope was manually voided or expired.
+* `ENVELOPE_DECLINED` - The envelope has been declined by one of the recipients.
+* `ENVELOPE_COMPLETED` - The envelope has been completed by all of the recipients.
+* `ENVELOPE_DELIVERED` - This event is sent when all recipients have opened the envelope through the DocuSign signing website. This does not signify an email delivery of an envelope.
+* `ENVELOPE_SENT` - This event is sent when the email notification, with a link to the envelope, is sent to at least one recipient or when it is a recipient's turn to sign during embedded signing. The envelope remains in this state until all recipients have viewed the envelope.
+
```

Comment:
These events can be consumed by the process using Triggers. For example, a BPMN catch message event inside a process can be waiting for a trigger event set to ENVELOPE_DECLINED and a trigger action set to a throw BPMN message.

Reply:
added **Note:** These events can be consumed by the process using Triggers. For example, a BPMN catch message event inside a process can be waiting for a trigger event set to ENVELOPE_DECLINED, and a trigger action set to throw a BPMN message.